Modules/Devices/src/Devices.Domain/Entities/Identities/Identity.cs

```diff
@@ -40,7 +41,19 @@ public Identity(string? clientId, IdentityAddress address, byte[] publicKey, Tie
     public byte IdentityVersion { get; private set; }
 
     public TierId? TierIdBeforeDeletion { get; private set; }
-    public TierId TierId { get; private set; }
+
+    public TierId TierId
+    {
+        get => _tierId;
+        private set
+        {
+            if (_tierId == null! || value == _tierId) return; //TODO: Timo (Since I use a backing field and init this one in the constructor, I don't need the null check, right?)
```

Comment:
I would still keep the null check. And use the property in the constructor.
Because what keeps someone in the future from using the property instead? Your code should be as fool proof as possible.